resources/guidelines/testing/store/quality-guidelines-plugins/README.md

```diff
@@ -0,0 +1,644 @@
+
+For more information, refer to [Shopware app guidelines](https://developers.shopware.com/developers-guide/plugin-guidelines/#plugin-installation-routine)
+
+### Shopping worlds/shopping experiences
+
+Shopping worlds elements must include an element icon. If the app is deleted, *Shopping Worlds* should continue to work flawlessly in the frontend.
+
+### Payment apps
+
+We check if the "pluginlogger" service is used for the debug/error.log and that logs are written in the directory /var/log/. Log files must use this folder in every circumstance. Another solution is to store them in the database.
+
+### Every app accessing external API services
+
+A test button for optional API access data must be available. If the API data is incorrect, an entry must appear in the event log file in the Shopware folder /var/log/ respectively in the database.
+
+Apart from that, you can validate the required credentials while saving them in the app settings. In this case, a status message must be displayed in the backend and Shopware log.
+
+[Example for implementing an API Test Button into the System Config form](https://github.com/shyim/ShyimApiTest)
+
+### Automatic code reviews with PhpStan and SonarQube
+
+Our most current code review configurations that we use when uploading apps via the Shopware Account can be found on GitHub.
+
+Refer to [code reviews for Shopware 5 on GitHub](https://github.com/shopwareLabs/store-plugin-codereview)
+
+### Automated code tests with Cypress
+
+There are Cypress tests for Shopware 5 and 6 on GitHub. The project is driven by the Friends of Shopware group. You can contribute at any time:
+
+*   [Cypress Tests for Shopware 5](https://github.com/FriendsOfShopware/FroshCypress)
+
+### Helpful tools for app developers
+
+* [FroshPluginUploader](https://github.com/FriendsOfShopware/FroshPluginUploader): Tool for validating and uploading new SW6 app releases to the Community Store (GitHub Project from "Friends of Shopware")]
+* [Shopware CLI tools](https://github.com/shopwareLabs/sw-cli-tools): When you think about performance, these are various useful console helpers for generating data.
+
+### Shopware 5.0 SDK License
+
+The Shopware 5 backend is completely based on the Ext JS framework by Sencha ([www.sencha.com](http://www.sencha.com/)). Suppose you are developing Shopware 5 apps that are built upon or alter the Shopware/ExtJS layer and wish to offer them with a proprietary license (apps that are not licensed under the GNU Affero General Public License version 3 or a compatible license); you require a Shopware SDK license.
+
+Thanks to an OEM agreement with Sencha, Shopware is able to offer this license on a per developer basis, allowing the license holder to use this framework for their own apps.
+
+**When will I not need a SW5 SDK-License?**
```

Comment:
Is it "when do i need a SW5 SDK license?" or "when i will not need a SW5 SDK license?"

Reply:
hmm, there is missing a part. Normally it would be like this:

When do I need a SW5 SDK-License?

1. Apps will be sold and uses extJS

When do I NOT need a SW5 SDK-License?

1. App will be sold and uses no extJS

2. App is for free and uses extJS

3. App is for free and uses no extJS

If you need a license, please write an email to alliances@shopware.com